# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1214510391 14400
#      Thu Jun 26 15:59:51 2008 -0400
# Node ID cce16b4e09702152c298fd78925ed6f4f202ec1d
# Parent  322a066543aca13a888e872ce388c8a26c1e5d16
lex.l (grab_block_comment): Use parens around || expression within && expression

diff --git a/src/lex.l b/src/lex.l
--- a/src/lex.l
+++ b/src/lex.l
@@ -1287,17 +1287,17 @@ grab_block_comment (stream_reader& reade
 		      buf += tmp_buf;
 		      done = true;
 		      break;
 		    }
 		}
 	    }
 	}
 
-      if (at_bol && c == '%' || c == '#')
+      if (at_bol && (c == '%' || c == '#'))
         {
           if (c == '#' && ! warned_incompatible)
 	    {
 	      warned_incompatible = true;
 	      maybe_gripe_matlab_incompatible_comment (c);
 	    }
 
 	  at_bol = false;
